api.md

```diff
@@ -919,6 +922,92 @@ The expected output:
 
 ---
 
+## interpolate() [](interpolate)
+
+The `interpolate` function does linear interpolation for missing values.
+It can only be used in an aggregation query with [time_bucket_gapfill](#time_bucket_gapfill).
+The `interpolate` function call cannot be nested inside other function calls.
+
+#### Required Arguments [](interpolate-required-arguments)
+
+|Name|Description|
+|---|---|
+| `value` | The value to interpolate (int2/int4/int8/float4/float8) |
+
+#### Optional Arguments [](interpolate-optional-arguments)
+
+|Name|Description|
+|---|---|
+| `prev` | The lookup expression for values before the gapfill time range (record) |
+| `next` | The lookup expression for values after the gapfill time range (record) |
+
+Because the interpolation function relies on having values before and after
+each bucketed period to compute the interpolated value, it might not have
+enough data to calculate the interpolation for the first and last time bucket
+if those buckets do not otherwise contain valid values.
+For example, the interpolation would require looking before this first
+time bucket period, yet the query's outer time predicate WHERE time > ...
+normally restricts the function to only evaluate values within this time range.
+Thus, the `prev` and `next` expression tell the function how to look for
+values outside of the range specified by the time predicate.
+These expressions will only be evaluated when no suitable value is returned by the outer query.
+This happens when the first and / or last bucket in the queried time range is empty.
+The returned record for `prev` and `next` needs to be a time, value tuple.
```

Comment:
I am sooo sorry for the additional comment, could we move "The returned record for `prev` and `next` up to `interpolate` call to go right after the optional arguments table? I think it's nice to say what type those arguments are right up front.

Reply:
Although now that I read it again, I'm not longer convinced. What do you think?

Reply:
Not sure I follow @dianasaur323 's comment.  One grammar nit though, perhaps:

`These expressions will only be evaluated when no suitable value is returned by the outer query.
This happens when the first and / or last bucket in the queried time range is empty.`

should be changed to:

`These expressions will only be evaluated when no suitable value is returned by the outer query (i.e., the first and/or last bucket in the queried time range is empty).`

as otherwise a bit repetitive.

Reply:
since mike prefers current version i left the order